### ISSUE TITLE
FS init causes memory allocation error (possible corruption)

### ISSUE THREAD

**Issue by 0x90nz** [0x90nz/microboot#27]
On GCC 11 and QEMU 6.1 there seems to be an issue where during FS init, some write occcurs which overwrites the magic signature on the heap.

Would be nice to have backtrace to debug this, so depends on #21.

**Comment by 0x90nz** [comment]
Yep. This is corruption, specifically caused by not allocating enough space here:
https://github.com/0x90nz/microboot/blob/194baa224b2472f0e5d3b31be4dae88c9e2edf4f/kern/fs/ext2.c#L322
(the inode might be 256 bytes, but the buffer is only ever 128)

Realistically I just need to rip EXT2 out and replace it with something simpler and more in line with what this project is about (see #30).

**Comment by 0x90nz** [comment]
Removing EXT2 support means this bug no longer happens
